zk_prover/src/circuits/utils.rs

```diff
@@ -47,14 +68,38 @@ pub fn generate_setup_params(k: u32) -> ParamsKZG<Bn256> {
         let timer = start_timer!(|| "Creating params");
         let params = ParamsKZG::<Bn256>::setup(k, OsRng);
         end_timer!(timer);
-        params
+        if params.k() < k {
```

Comment:
In which case this error would get triggered here? 

While I see this error being possible in the `else` case. Here it seems to me that the `.k()` of params will always be equal to the `k` that we passed as input as `params` are generated using `k`.

Reply:
True, no need this check on that. I will remove it.

Reply:
Fixed at [42b1dba](https://github.com/summa-dev/summa-solvency/pull/117/commits/42b1dbad462239eb315390299bf33e84148aa294)